cmd/env.go

```diff
@@ -136,55 +145,56 @@ var envAddCmd = &cobra.Command{
 		return c.Run()
 	},
 
-	Long: `Add a new environment to a ksonnet project. Names are restricted to not
-include punctuation, so names like` + " `../foo` " + `are not allowed.
-
-An environment acts as a sort of "named cluster", allowing for commands like
-` + " `ks apply dev` " + `, which applies the ksonnet application to the "dev cluster".
-For more information on what an environment is and how they work, run` + " `ks help env` " + `.
-
-Environments are represented as a hierarchy in the 'environments' directory of a
-ksonnet application, and hence` + " `ks env add` " + `will add to this directory structure.
-For example, in the example below, there are two environments: 'default' and
-'us-west/staging'.` + " `ks env add` " + `will add a similar directory to this environment.
-
-    environments/
-      default/           [Default generated environment]
-        .metadata/
-          k.libsonnet
-          k8s.libsonnet
-          swagger.json
-        spec.json
-		default.jsonnet
-        params.libsonnet
-      us-west/
-        staging/         [Example of user-generated env]
-          .metadata/
-            k.libsonnet
-            k8s.libsonnet
-            swagger.json
-          spec.json      [This will contain the API server address of the environment and other environment metadata],
-		  staging.jsonnet
-          params.libsonnet`,
-	Example: `# Initialize a new staging environment at 'us-west'.
-# The environment will be setup using the current context in your kubecfg file. The directory
-# structure rooted at 'us-west' in the documentation above will be generated.
+	Long: `
+The ` + "`add`" + ` command creates a new environment (specifically for the ksonnet app
+whose directory it's executed in). This environment is cached with the following
+info:
+
+1. **Name** — A string used to uniquely identify the environment.
+2. **Server** — The address and port of a Kubernetes API server (i.e. cluster).
+3. **Namespace**  — A Kubernetes namespace. *Must already exist on the cluster.*
+4. **Kubernetes API Version**  — Used to generate a library with compatible type defs.
+
+(1) is mandatory. (2) and (3) can be inferred from $KUBECONFIG, *or* from the
+` + "`--kubeconfig`" + ` or ` + "`--context`" + ` flags. Otherwise, (2), (3), and (4) can all be
+specified by individual flags.
+
+Note that an environment *DOES NOT* contain user-specific data such as private keys.
+
```

Comment:
Worth calling out that environments are initialized based on a specific version of Kubernetes and that the latest that ksonnet supports is used unless specified otherwise.